tests/test_quantities.py

```diff
@@ -801,7 +801,7 @@ def test_quantity_divided_by_float(
 ) -> None:
     """Test the division of all quantities by a float."""
     hypothesis.assume(scalar != 0.0)
-    quantity = quantity_ctor(quantity_value)
+    quantity = quantity_ctor(quantity_value)  # pylint: disable=unreachable
```

Comment:
Really? This feels like a regression in hypothesis, probably the added `-> NoReturn` to `assume()` or something like that. Super weird.

I didn't find a issue in hypothesis, but found other projects with a very recent similar issue:

* https://github.com/apache/beam/issues/36951

Since these updates are not urgent, I'm more inclined to wait a bit to see if it gets resolved. Maybe we should report upstream.